src/Form/IslandoraRiprapSettingsForm.php

```diff
@@ -24,56 +25,158 @@ protected function getEditableConfigNames() {
+      '#default_value' => $config->get('fixity_content_type') ? $config->get('fixity_content_type') : 'Repository Item',
+    ];
+    $form['fixity_terms'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Media Use terms'),
+      '#description' => $this->t('Media tagged with these terms will be checked. Seperate terms with commas.'),
+      '#default_value' => $config->get('fixity_terms') ? $config->get('fixity_terms') : array_search('Original File', $term_data),
+      '#options' => $term_data,
+    ];
+    $form['user_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Drupal user'),
+      '#description' => $this->t('User name with media access permsissions'),
+      '#default_value' => $config->get('user_name') ? $config->get('user_name') : '',
+    ];
+    $form['user_pass'] = [
+      '#type' => 'password',
+      '#title' => $this->t('Drupal password'),
+      '#description' => $this->t('Password for user with media access permsissions'),
+      '#default_value' => $config->get('user_pass') ? $config->get('user_pass') : '',
+    ];
+
+    $form['config'] = [
+      '#type' => 'details',
+      '#collapsible' => TRUE,
+      '#collapsed' => TRUE,
+      '#title' => t('Current configuration'),
+      ];
+    $form['config']['current setup'] = [
+      '#description' => t('Username and password ha\ve been obfuscated'),
+      '#markup' => $current_config,
+      '#title' => t('Current configuration'),
+    ];
 
     return parent::buildForm($form, $form_state);
   }
 
-  /** 
+  /**
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-       $this->configFactory->getEditable('islandora_riprap.settings')
-      ->set('riprap_rest_endpoint', $form_state->getValue('riprap_rest_endpoint'))
+    $values = $form_state->getValues();
+    $this->persistConfig($values);
+    $this->configFactory->getEditable('islandora_riprap.settings')
+      ->set('riprap_rest_endpoint', rtrim($form_state->getValue('riprap_rest_endpoint'), '/'))
       ->set('number_of_events', $form_state->getValue('number_of_events'))
-      ->set('gemini_rest_endpoint', $form_state->getValue('gemini_rest_endpoint'))
+      ->set('gemini_rest_endpoint', rtrim($form_state->getValue('gemini_rest_endpoint'), '/'))
       ->set('use_drupal_urls', $form_state->getValue('use_drupal_urls'))
       ->set('show_riprap_warnings', $form_state->getValue('show_riprap_warnings'))
+      ->set('fixity_terms', $form_state->getValue('fixity_terms'))
+      ->set('fixity_content_type', $form_state->getValue('fixity_content_type'))
+      ->set('user_name', $form_state->getValue('user_name'))
+      ->set('user_pass', $form_state->getValue('user_pass'))
       ->save();
 
     parent::submitForm($form, $form_state);
   }
+
+  public function persistConfig($values) {
+    $base_url = \Drupal::request()->getSchemeAndHttpHost();
+    $config_file_path = "public://rdm_config";
```

Comment:
I don't mind having an unencrypted password in the admin settings for this module, but writing a file to the public filesystem that contains credentials is not good. Can we stuff the contents of the config file in an admin setting that doesn't have any widgets in the admin UI? I think creating a separate db table for this is overkill.